mapeditor/mapsettings.cpp

```diff
@@ -108,5 +454,423 @@ void MapSettings::on_pushButton_clicked()
+				specialDefeat.trigger = EventExpression(noneOf);
+				break;
+			}
+				
+			case 1: {
+				EventExpression::OperatorNone noneOf;
+				EventCondition cond(EventCondition::CONTROL);
+				cond.objectType = Obj::HERO;
+				assert(loseTypeWidget);
+				int townIdx = loseTypeWidget->currentData().toInt();
+				cond.position = controller.map()->objects[townIdx]->pos;
+				noneOf.expressions.push_back(cond);
+				specialDefeat.onFulfill = VLC->generaltexth->allTexts[253];
+				specialDefeat.trigger = EventExpression(noneOf);
+				break;
+			}
+				
+			case 2: {
+				EventCondition cond(EventCondition::DAYS_PASSED);
+				assert(loseValueWidget);
+				cond.value = expiredDate(loseValueWidget->text());
+				specialDefeat.onFulfill = VLC->generaltexth->allTexts[254];
+				specialDefeat.trigger = EventExpression(cond);
+				break;
+			}
+				
+			case 3: {
+				EventCondition cond(EventCondition::DAYS_WITHOUT_TOWN);
+				assert(loseValueWidget);
+				cond.value = loseValueWidget->text().toInt();
+				specialDefeat.onFulfill = VLC->generaltexth->allTexts[7];
+				specialDefeat.trigger = EventExpression(cond);
+				break;
+			}
+		}
+		
+		EventExpression::OperatorAll allOf;
+		EventCondition isHuman(EventCondition::IS_HUMAN);
+		isHuman.value = 1;
+
+		allOf.expressions.push_back(specialDefeat.trigger.get());
+		allOf.expressions.push_back(isHuman);
+		specialDefeat.trigger = EventExpression(allOf);
+
+		if(ui->standardLoseCheck->isChecked())
+		{
+			controller.map()->triggeredEvents.push_back(standardDefeat);
+		}
+		controller.map()->triggeredEvents.push_back(specialDefeat);
+	}
+	
 	close();
 }
+
+void MapSettings::on_victoryComboBox_currentIndexChanged(int index)
+{
+	delete victoryTypeWidget;
+	delete victoryValueWidget;
+	delete victorySelectWidget;
+	victoryTypeWidget = nullptr;
+	victoryValueWidget = nullptr;
+	victorySelectWidget = nullptr;
+	
+	if(index == 0)
+	{
+		ui->standardVictoryCheck->setChecked(true);
+		ui->standardVictoryCheck->setEnabled(false);
+		ui->onlyForHumansCheck->setChecked(false);
+		ui->onlyForHumansCheck->setEnabled(false);
+		return;
+	}
+	ui->onlyForHumansCheck->setEnabled(true);
+	ui->standardVictoryCheck->setEnabled(true);
+	
+	int vicCondition = index - 1;
+	switch(vicCondition)
+	{
+		case 0: { //EventCondition::HAVE_ARTIFACT
+			victoryTypeWidget = new QComboBox;
+			ui->victoryParamsLayout->addWidget(victoryTypeWidget);
+			for(int i = 0; i < controller.map()->allowedArtifact.size(); ++i)
+				victoryTypeWidget->addItem(QString::fromStdString(VLC->arth->objects[i]->getNameTranslated()), QVariant::fromValue(i));
+			break;
+		}
+			
+		case 1: { //EventCondition::HAVE_CREATURES
+			victoryTypeWidget = new QComboBox;
+			ui->victoryParamsLayout->addWidget(victoryTypeWidget);
+			for(int i = 0; i < VLC->creh->objects.size(); ++i)
+				victoryTypeWidget->addItem(QString::fromStdString(VLC->creh->objects[i]->getNameTranslated()), QVariant::fromValue(i));
```

Comment:
Creatures have singular & plural name, so `getNameTranslated` is explicitly removed for them
```suggestion
				victoryTypeWidget->addItem(QString::fromStdString(VLC->creh->objects[i]->getNameSingularTranslated()), QVariant::fromValue(i));
```

Reply:
Or another option, depending on context - use plural name:
```suggestion
				victoryTypeWidget->addItem(QString::fromStdString(VLC->creh->objects[i]->getNamePluralTranslated()), QVariant::fromValue(i));
```